### PR TITLE
Modified the way to add singleton resources.

### PR DESCRIPTION
**Pull request by olalonde**
See README.md. I think it's much cleaner now. Mapping the singleton resource to a controller shouldn't be railway-routes' job since it's not expected as you mentioned. Let me know if you like this version better. I hope I didn't forget to follow the conventions too many times :/

### CODE REVIEW

File: examples/singleton/server.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+var app = express();
+
+app.use(app.router);
+
+var map = new Map(app, function(namespace, controller, action) {
+  console.log(arguments);
+});
+
+map.resources('users', function(user) {
+  map.resources('posts', function(post) {
```
Comment (anatoliychakkaev) [inline examples/singleton/server.js:13]: I guess it should be `user.resources('posts',...`?

Comment (olalonde) [inline examples/singleton/server.js:13]: Oops yeah :/

**Comment by olalonde** [comment]
Hey I fixed the README and added another commit to implement something I needed. Let me explain what I'm trying to achieve first. In my app, there are 3 ways to access a user's profile.

/account (user profile of currently logged user)
/users/:user_id (any user's profile using user_id)
/:username (any user's profile using username)

All those routes should have the same nested resources (subroutes) as they all basically represent the same resource. Here's how I implemented it:

```
var addUserSubresources = function(user) {
  user.resources('posts');
};

map.resources('users', addUserSubresources);
map.resource('account', addUserSubresources);
map.resource('users' , {path: ':username'}, addUserSubresources);
```

ps: thinking about it now, I should actually replace `map.resource('account', addUserSubresources);` by `map.resource('users', {path: 'account'}, addUserSubresources);`

Anyways, the problem was that the nested routes path always begun with /users/. So I think I fixed this in this commit by prefixing subroutes by params.path if it is available, otherwise default to resource name like before.

It would also be nice if I could do something like below and I may work on implement it if you like the idea.

```
map.resources('users', {paths: [':username', 'account']}, function(user) {
  user.resources('posts');
});
```

Let me know what you think about this.

**Comment by olalonde** [comment]
Oops, forgot to remove my console.log()-foo :)

**Pull request merged**